scio-google-cloud-platform/src/it/scala/com/spotify/scio/bigquery/BigQueryIT.scala

```diff
@@ -58,15 +58,23 @@ class BigQueryIT extends AnyFlatSpec with Matchers {
       shakespeare("thy", 30, "kinglear", 1600)
     )
     val expected = Seq(wordCount("i", 10), wordCount("thou", 20), wordCount("thy", 30))
+  }
 
+  "MockBigQuery" should "support mock data" in {
     val mbq = MockBigQuery()
-    mbq.mockTable(tableRef).withData(inData)
-    mbq.queryResult(legacyQuery) should contain theSameElementsAs expected
-    mbq.queryResult(sqlQuery) should contain theSameElementsAs expected
+    mbq.mockTable(tableRef).withData(MockBQData.inData)
+    mbq.queryResult(legacyQuery) should contain theSameElementsAs MockBQData.expected
+    mbq.queryResult(sqlQuery) should contain theSameElementsAs MockBQData.expected
   }
 
-  // see https://cloud.google.com/bigquery/docs/querying-wildcard-tables
-  it should "support wildcard tables" in {
+  "MockBigQuery" should "support mock data with custom staging dataset" in {
+    val mbq = MockBigQuery(stagingDataset = Some("scio_bigquery_staging_custom"))
+    mbq.mockTable(tableRef).withData(MockBQData.inData)
+    mbq.queryResult(legacyQuery) should contain theSameElementsAs MockBQData.expected
+    mbq.queryResult(sqlQuery) should contain theSameElementsAs MockBQData.expected
```

Comment:
Isn't this querying the data from the previous test ?
We should have custom queries with the overridden dataset here

Reply:
can you help me understand how it is different from before? We used to have two tests (mock and wildcard mock) both hitting the same staging dataset `scio_bigquery_staging_us`, and now both of them are duplicated to run the same tests on the new parallel dataset `scio_bigquery_staging_custom`

Reply:
Ok, my bad.
I was confused that the query contains a dataset which is not the same as the one given to the `mbq` but this is actually what we expect -> the tableref is swapped with one in the `stagingDataset`

However, how do you actually make sure in this test that the provided `stagingDataset` is used instead of the default one?

Reply:
If it is not too hard, we can try to issue an extra BQ describe command on the dataset/table to be sure it is there.
Otherwise I trust the good implementation :)